braindecode/datasets/siena.py

```diff
@@ -0,0 +1,198 @@
+"""
+This dataset is a BIDS compatible version of the Siena Scalp EEG Database.
+
+It reorganizes the file structure to comply with the BIDS specification. To this effect:
+
+-   Metadata was organized according to BIDS.
+-   Data in the EEG edf files was modified to keep only the 19 channels from a 10-20 EEG system.
+-   Annotations were formatted as BIDS-score compatible tsv files.
+
+"""
+
+# Authors: Dan, Jonathan
+#          Detti, Paolo
+#          Bruno Aristimunha <b.aristimunha@gmail.com>
+#
+# License: BSD (3-clause)
+from __future__ import annotations
+
+import glob
+import os
+from pathlib import Path
+
+from mne.datasets import fetch_dataset
+
+from braindecode.datasets import BIDSDataset
+
+SIENA_URL = "https://zenodo.org/records/10640762/files/BIDS_Siena.zip"
+SIENA_archive_name = "SIENA.zip"
+SIENA_folder_name = "SIENA-BIDS-eeg-dataset"
+SIENA_dataset_name = "SIENA-EEG-Corpus"
+
+SIENA_dataset_params = {
+    "dataset_name": SIENA_dataset_name,
+    "url": SIENA_URL,
+    "archive_name": SIENA_archive_name,
+    "folder_name": SIENA_folder_name,
+    "hash": "126e71e18570cf359a440ba5227494ecffca4b0b0057c733f90ec29ba5e15ff8",  # sha256
+    "config_key": SIENA_dataset_name,
+}
+
+
+class SIENA(BIDSDataset):
+    """The Siena EEG Dataset.
+
+    The database consists of EEG recordings of 14 patients acquired at the Unit of Neurology
+    and Neurophysiology of the University of Siena.
+
+    Subjects include 9 males (ages 25-71) and 5 females (ages 20-58).
+    Subjects were monitored with a Video-EEG with a sampling rate of 512 Hz,
```

Comment:
In the BIDS version of the dataset, which is used here, the sample rate is fixed to 250Hz